src/ucs/vfs/fuse/vfs_fuse.c

```diff
@@ -330,6 +332,15 @@ static ucs_status_t ucs_vfs_fuse_wait_for_path(const char *path)
         for (offset  = 0; offset < nread;
              offset += (sizeof(*event) + event->len)) {
             event = UCS_PTR_BYTE_OFFSET(event_buf, offset);
+
+            /* Watch was removed explicitly (inotify_rm_watch) or automatically
+             * (file was deleted, or file system was unmounted). */
+            if (event->mask & IN_IGNORED) {
+                ucs_debug("inotify watch on '%s' was removed", watch_dirname);
```

Comment:
shall we check stop flag?

Reply:
no, we shouldn't
Because we check it already on line 317 - in that case no matter what's event - we cancel the waiting.
Then we remain holding the mutex - so stop flag is not possible to set.
Here we are guaranteed that `stop` flag is not set, so we mostly check if the folder was removed from the filesystem (means implicitly, not from VFS)

Btw, regarding stop flag, we can also simplify the code above: move lines 294-302 to be right after mutex acquisition, so that when `stop` flag is set we detect it early and don't even bother creating folder, initializing inotify etc

Reply:
maybe we should try re-creating the watch in case it was removed?

Reply:
I was thinking about this, and even implemented (and tested) this approach, it's really just a matter of 2 LOCs. Then I decided to not go this way, because:
- deletion of socket folder is not a part of the normal workflow, and it is not handled gracefully by VFS daemon. Daemon does not re-establish a socket listen if it gets deleted
- continue reconnection effort means we need to re-create the folder, try to connect to socket (and then redo watch) - this may consume CPU resources
- Currently function `ucs_vfs_sock_mkdir` that we use to create a folder, does not create a path (like `mkdir -p`), but creates just the last folder from path. So if the whole user folder is deleted (like happens with `mtt` user), we cannot recreate this path anyway

So I thought that it's not worth to implement re-connection in this case. But I don't have strong opinion on that, please tell me if you do

Reply:
ok, let'd keep as is, i guess if the folder is removed it's not a case we have to support